src/hotspot/share/gc/g1/g1MonitoringSupport.hpp

```diff
@@ -242,8 +241,28 @@ class G1MonitoringScope : public StackObj {
   TraceCollectorStats _tcs;
   TraceMemoryManagerStats _tms;
 public:
-  G1MonitoringScope(G1MonitoringSupport* monitoring_support, bool full_gc, bool all_memory_pools_affected);
+  G1MonitoringScope(G1MonitoringSupport* monitoring_support,
+                    CollectorCounters* collection_counters,
+                    GCMemoryManager* gc_memory_manager,
+                    bool all_memory_pools_affected = true);
   ~G1MonitoringScope();
 };
 
+class G1YoungGCMonitoringScope : public G1MonitoringScope {
+public:
+  G1YoungGCMonitoringScope(G1MonitoringSupport* monitoring_support, bool all_memory_pools_affected);
+  ~G1YoungGCMonitoringScope() { }
```

Comment:
I think the empty destructor in child classes can be omitted, and the destructor in base class can probably be `protected`.

Reply:
done